python/tests/unit/arcticdb/version_store/test_append.py

```diff
@@ -16,6 +17,8 @@
 from arcticdb_ext.exceptions import InternalException, NormalizationException, SortingException
 from arcticdb_ext import set_config_int
 
+random.seed(42)
```

Comment:
It is more beneficial for the tests to be random even if they're a bit flaky. We found so many bugs by having random data.

Reply:
I added a check to seed them only in a specific case, by default we will not seed them.
I need the seed to use [pytest-xdist](https://pytest-xdist.readthedocs.io/en/latest/known-limitations.html) to parallelize the tests.

Reply:
I still failed to see why `pytest-xdist` would require you to use a fixed seed?

Reply:
From the xdist documentation
```
Order and amount of test must be consistent[](https://pytest-xdist.readthedocs.io/en/latest/known-limitations.html#order-and-amount-of-test-must-be-consistent)
It is not possible to have tests that differ in order or their amount across workers.
```

Because the parameters are part of each individual test signature, xdist thinks that different workers are seeing different test cases.

Reply:
So this is to make those `gen_param...` functions work?

That is a pretty bad pattern anyway, given it will break test framework integration in all kinds of tooling.

Could you log an issue to replace those with Hypothesis?

Also, xdist only needs these randomly generated test parameters to be the same across all forks, so you can set an environment variable to a timestamp and then seed with that in Python. This will at least allow us to get different test cases from different runs of the job.

It will also be more clear to only fix the seed in those `gen_param...` functions and reset afterwards. You can easily write a decorator to do that.

Reply:
yeah that makes sense, I will try it